cmd/tempo-serverless/handler.go

```diff
@@ -110,7 +110,7 @@ func Handler(r *http.Request) (*tempopb.SearchResponse, *HTTPError) {
 	var resp *tempopb.SearchResponse
 
 	if api.IsTraceQLQuery(searchReq.SearchReq) {
-		engine := traceql.NewEngine()
+		engine := traceql.NewEngineWithSpansPerSpanSet(int(searchReq.SearchReq.SpansPerSpanSet))
```

Comment:
Let's have `NewEngine()` take the spans per spanset parameter instead of making a new method. There are 2 places in `instance_search` that will also need this parameter for it to work.

Reply:
I see three invocations of NewEngine in instance_search, 2 of which has access to request. Third one desn't have, so I guess I can pass default value of 3 there.

Reply:
I looked a little deeper and I'm wondering why we don't just use the value on the `searchReq`? We pass it in when we call `ExecuteSearch`. Can't we just the value on that struct?

Reply:
You mean to add spanperspanset filed in SearchBlockRequest instead of SearchRequest?